variables.tf

```diff
@@ -271,6 +284,24 @@ EOF
 
 }
 
+variable "http_proxy" {
+  type        = string
+  default     = ""
+  description = <<EOF
+HTTP proxy for downloading agents and vault executable on startup. Only necessary if allow_public_egress is false.
```

Comment:
Maybe emphasize that this is _only_ for the startup script. This doesn't set a global HTTP_PROXY envvar.